cli/cli.go

```diff
@@ -2,43 +2,105 @@ package cli
 
 import (
 	"fmt"
-	"github.com/shpota/goxygen/codegen"
 	"io"
 	"regexp"
+	"strings"
+
+	"github.com/shpota/goxygen/codegen"
 )
 
 func Start(w io.Writer, commands []string) {
-	if len(commands) == 1 && commands[0] == "help" {
+	argsLen := len(commands)
+	if argsLen == 1 && commands[0] == "help" {
 		fmt.Fprintln(w, usage)
 		return
 	}
-	if len(commands) == 2 && commands[0] == "init" {
-		projectName := commands[1]
-		validName, _ := regexp.MatchString("^[a-zA-Z0-9_-]+$", projectName)
-		if validName {
-			codegen.Generate(projectName)
+	if (argsLen == 2 || argsLen == 3) && commands[0] == "init" {
+		var frontendTech string
+		var projectName string
+
+		projectName, ok, errMsg := extractProjectName(commands[argsLen-1])
+		if !ok {
+			fmt.Fprintln(w, errMsg)
 			return
 		}
-		fmt.Fprintln(w, invalidName)
+		if argsLen == 2 {
+			codegen.Generate(projectName, strings.ToLower(React))
+			return
+		}
+
+		frontendTech, ok, errMsg = extractFrontendTechName(commands[1])
+		if !ok {
+			fmt.Fprintln(w, errMsg)
+			return
+		}
+		codegen.Generate(projectName, strings.ToLower(frontendTech))
 		return
 	}
-	fmt.Fprintln(w, wrongInput)
+	fmt.Fprintln(w, wrongInputErr)
+	return
 }
 
 const usage = `Usage:
 
-   go run github.com/shpota/goxygen init <project-name>
+	go run github.com/shpota/goxygen init [--frontend] <project-name>
 
 Generates a web project with the following structure:
-server side Go code, webapp with corresponding React
-components, a Dockerfile that builds the application,
-and Docker Compose scripts to run the app in dev and
-prod modes.`
+server side Go code, webapp with corresponding frontend
+components, by default React, a Dockerfile that builds 
+the application,and Docker Compose scripts to run the app
+in dev andprod modes.`
 
-const wrongInput = `Wrong input! Use:
-   go run github.com/shpota/goxygen init <project-name>
+const wrongInputErr = `Wrong input! Use:
+   go run github.com/shpota/goxygen init [--frontend] <project-name>
 In order to get more details run:
    go run github.com/shpota/goxygen help`
 
-const invalidName = `Project name is not valid. The allowed symbols are letters,
-numbers, underscores, and dashes.`
+const missProjectNameErr = `Project Name missing!`
+
+const Vue = "Vue"
+const Angular = "Angular"
+const React = "React"
+const flags = "--"
+
+func invalidProjectNameErr(name string) string {
+	return fmt.Sprintf(`Project name %v is not valid. The allowed symbols are letters,
+numbers, underscores, and dashes.`, name)
+}
+
+func invalidFrontendTechErr(name string) string {
```

Comment:
I guess we don't need such a precision in error messages. If a user enters an invalid input it is enough to show them a generalized error message with an example on how to run it. It will be simpler from development perspective and while still being helpful for users.